CompatibilityShapeData was not properly scoped so `deprecated=no` does not compile.

**Godot version:**
c3f7465b7efe233eaa8945e41f4029840c1aa153

**OS/device including version:**
win10

**Issue description:**
https://github.com/godotengine/godot/blob/c3f7465b7efe233eaa8945e41f4029840c1aa153/scene/resources/tile_set.h#L68

CompatibilityShapeData was not properly scoped so `deprecated=no` does not compile.

https://github.com/godotengine/godot/blob/c3f7465b7efe233eaa8945e41f4029840c1aa153/scene/resources/tile_set.cpp#L748-L756 

When was compatibility deprecation added? Can we delete the old code?

**Steps to reproduce:**
```
scons platform=windows -j8 bits=64 dev=yes deprecated=no target=debug
```

**Minimal reproduction project:**
See above.

`dev=yes` fails for other reasons including this issue, but we can remove it.

The log is for:

```scons platform=windows -j8 bits=64 deprecated=no target=debug```

```
scene\2d\tile_map.cpp(827): error C2039: 'compatibility_get_source_for_tile_id': is not a member of 'TileSet'
C:\Users\ernest.lee\Desktop\godot\scene/resources/tile_set.h(65): note: see declaration of 'TileSet'
```

--- a/scene/2d/tile_map.cpp
+++ b/scene/2d/tile_map.cpp
@@ -777,6 +777,11 @@ void TileMap::_set_tile_data(const Vector<int> &p_data) {
 	int offset = (format >= FORMAT_2) ? 3 : 2;
 
 	clear();
+
+#ifdef DISABLE_DEPRECATED
+	ERR_FAIL_COND_MSG(format != FORMAT_3, vformat("Cannot handle deprecated TileMap data format version %d. This Godot version was compiled with no support for deprecated data.", format));
+#endif
+
 	for (int i = 0; i < c; i += offset) {
 		const uint8_t *ptr = (const uint8_t *)&r[i];
 		uint8_t local[12];
@@ -806,6 +811,7 @@ void TileMap::_set_tile_data(const Vector<int> &p_data) {
 			uint16_t alternative_tile = decode_uint16(&local[10]);
 			set_cell(Vector2i(x, y), source_id, Vector2i(atlas_coords_x, atlas_coords_y), alternative_tile);
 		} else {
+#ifndef DISABLE_DEPRECATED
 			uint32_t v = decode_uint32(&local[4]);
 			v &= (1 << 29) - 1;
 
@@ -828,6 +834,7 @@ void TileMap::_set_tile_data(const Vector<int> &p_data) {
 			}
 
 			set_cell(Vector2i(x, y), v, Vector2i(coord_x, coord_y), compatibility_alternative_tile);
+#endif
 		}
 	}
 }

